client/src/components/AddItem.js

```diff
@@ -0,0 +1,53 @@
+import { useState, useContext } from "react";
+import { TextField, Grid } from "@material-ui/core";
+
+import { MyContext } from "../Provider";
+import API from "../utils/API";
+import CusButton from "./CustomButton";
+
+export const AddItem = ({ setPosts, posts }) => {
+  let state = useContext(MyContext);
+  let [title, setTitle] = useState("");
+  let [description, setDescription] = useState("");
+  let [topics] = useState([]);
+
+  const addTask = () => {
+    API.addPost(
+      { title, description, topics },
+      state.tokenConfig()
+    ).then((result) => setPosts([...posts].concat(result.data)));
+  };
```

Comment:
Once again, extract this.
Also, I would name the component `AddTask`. That's more descriptive.